Design/Instrument-2.0/Detector_Scan_NeXus_Saving.md

```diff
@@ -0,0 +1,129 @@
+├───instrument
+│   ├───detector_1
+│   │   ├───detector_number
+│   │   ├───x_pixel_offset
+│   │   ├───y_pixel_offset
+│   │   ├───z_pixel_offset
+│   │   ├───position_scan_x (attribute 'vector' for transformation, e.g. (1., 0., 0.))
+│   │   │   ├───scan_log
+│   │   │       ├───time (contains the time for each entry in value)
+│   │   │       ├───value (single number per time entry, describing the distance along 'vector')
+│   │   ├───position_scan_...
+│   ├───detector_2
+│   ├───...
+├───sample
+```
+
+An open question here, especially when considering conversion from the old IDFs, is how to define the component used for `detector_X`. Technically this could be anything from the whole instrument to individual pixels, but logically it should be something like a tube. The only requirement is that a detector has pixels with a fixed offset at all times.
+
+#### Differences Between NeXus and Mantid Approach
+
+In the `DetectorInfo` representation in Mantid each position, rotation and mask flag of each pixel at each time index is stored separately. The scans are done on a pixel-by-pixel basis, and the components can not currently do any scanning. Different pixels are allowed to be given different scan intervals even within a component.
+
+The NeXus instrument proposal would not support scanning at the pixel level. The offset entries would be the same for each time index, with the position entries giving the scan for a component.
+
+In the case of writing raw NeXus files from the instrument it is likely you would only need a subset of the offset, position scan and rotation entries based on physical knowledge of the instrument. After processing with Mantid this might not be so easy to achieve, as detectors can be moved around in an arbitrary way, e.g. for calibration.
+
+The NeXus proposal for scanning allows the start and end positions and rotations to be different during a scan interval. This is not supported in Mantid.
+
+### Component Scanning
+
+_Based on discussion with Simon Heybrock._
+
+To resolve the differences between the NeXus proposal and Mantid one option is to allow components to scan. This is similar to how `RectangularDetector` works. The changes required would be:
+
+* Add the scan interface in `DetectorInfo` to `ComponentInfo`
+  * E.g. in addition to `ComponentInfo::setPosition(detIndex)` we have `ComponentInfo::setPosition({detIndex, timeIndex})`
+  * A `ComponentInfo::setPosition()` call updates all of the child positions (same for rotations)
+* For setting positions, rotations or mask that can change with the time index, both `ComponentInfo` and `DetectorInfo` will require a check to see if the parent is scanning
+  * If the parent is scanning it should throw
+* The in memory representation is the same as before - the absolute position and rotation of all pixels are stored
+
+The performance impact of this is that every call to `DetectorInfo::position()` will have an extra branch, even for non-scanning workspaces.
```

Comment:
Would it? If we update positions when written you have extra branches in `setPosition()`, but wouldn't `position()` be unchanged?

Reply:
Yes, good point, that should be `setPosition()`.